aws-sdk-core/lib/aws-sdk-core/shared_config.rb

```diff
@@ -0,0 +1,179 @@
+module Aws
+  class SharedConfig
+
+    # @return [String]
+    attr_reader :credentials_path
+
+    # @return [String]
+    attr_reader :config_path
+
+    # @return [String]
+    attr_reader :profile_name
+
+    # Constructs a new SharedConfig provider object. This will load the shared
+    # credentials file, and optionally the shared configuration file, as ini
+    # files which support profiles.
+    #
+    # By default, only the shared credential file (the default path for which is
+    # `~/.aws/credentials`) is loaded. However, if you set the
+    # `ENV['AWS_SDK_LOAD_CONFIG']` environment variable, the shared config file
+    # will also be loaded (the default path for which is `~/.aws/config`).
```

Comment:
I would suggest moving the `ENV['AWS_SDK_LOAD_CONFIG']` completely outside the shared config class. Loading from the shared config should be the default behavior of this class. It could instead accept a boolean at construction time to disable this feature. The code above that constructs the default shared config object could do the environment check. This ensures defaults are preserved and eliminates the need for a process wide environment variable in applications that construct this directly.